app/Tasks/Start.php

```diff
@@ -12,16 +13,25 @@ class Start extends Task
     /** @var int $timeout */
     protected $timeout = 60; // seconds
 
+    /** @var bool $checks */
+    protected $checks = true;
+
     /** @var string $services */
     protected $services;
 
     public function run(...$args): int
     {
-        return $this->runCallables([
-            [$this, 'checkDependencies'],
-            [$this, 'dockerComposeUpD'],
-            [$this, 'database'],
-        ]);
+        $tasks = [
+            [$this, 'handleNetworkTask'],
+            [$this, 'startContainers'],
+        ];
+
+        if ($this->checks) {
+            array_unshift($tasks, [$this, 'checkDependencies']);
```

Comment:
maybe it's simpler if you just call `checkDependencies` and `checkDatabase` and within the method you just check; `if !checks return 0`?

Reply:
That is debatable, my line of thought:

Adding the check within the methods adds to their behaviour and responsibility, whereas each method has its proper goal and descriptive naming, adding such conditionals within could cause for unintended consequences: you can call a method checkDatabase, but there is something else that changes what it actually does?

So I rather avoid calling instead of conditionally behaving within the methods.